modules/base/include/modules/base/processors/vectortobuffer.h

```diff
@@ -0,0 +1,105 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/base/basemoduledefine.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/optionproperty.h>
+#include <inviwo/core/ports/bufferport.h>
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/datastructures/datatraits.h>
+
+namespace inviwo {
+
+template <typename T>
+class VectorToBuffer : public Processor {
+public:
+    VectorToBuffer()
+        : Processor()
+        , inport_{"inport"}
+        , outport_{"outport"}
+        , bufferUsage_{"bufferUsage",
+                       "Buffer Usage",
+                       {BufferUsage::Static, BufferUsage::Dynamic},
+                       0}
+        , bufferTarget_{
+              "bufferTarget", "Buffer Target", {BufferTarget::Data, BufferTarget::Index}, 0} {
+
+        addPorts(inport_, outport_);
+        addProperties(bufferUsage_, bufferTarget_);
+    }
+    virtual ~VectorToBuffer() override = default;
+
+    virtual void process() override {
+        if (bufferTarget_.getSelectedValue() == BufferTarget::Data) {
+            auto repr = std::make_shared<BufferRAMPrecision<T, BufferTarget::Data>>(
+                *inport_.getData(), bufferUsage_.getSelectedValue());
+            auto buffer = std::make_shared<Buffer<T, BufferTarget::Data>>(repr);
+            outport_.setData(buffer);
+        } else {
+            auto repr = std::make_shared<BufferRAMPrecision<T, BufferTarget::Index>>(
+                *inport_.getData(), bufferUsage_.getSelectedValue());
+            auto buffer = std::make_shared<Buffer<T, BufferTarget::Index>>(repr);
+            outport_.setData(buffer);
+        }
+    }
+
+    virtual const ProcessorInfo getProcessorInfo() const override;
+    static const ProcessorInfo processorInfo_;
+
+private:
+    DataInport<std::vector<T>> inport_;
+    BufferOutport outport_;
+
+    OptionProperty<BufferUsage> bufferUsage_;
+    OptionProperty<BufferTarget> bufferTarget_;
+};
+
+template <typename T>
+const ProcessorInfo VectorToBuffer<T>::getProcessorInfo() const {
+    return ProcessorTraits<VectorToBuffer<T>>::getProcessorInfo();
+}
+
+template <typename T>
+struct ProcessorTraits<VectorToBuffer<T>> {
+    static ProcessorInfo getProcessorInfo() {
+        return {
+            fmt::format("org.inviwo.{}.ToBuffer",
```

Comment:
In case of IndexBuffer, we probably want to name it `VectorToIndexBuffer`

Reply:
That is not possible, since not all integer ones will be IndexBuffers and the setting is at runtime.